test/e2e/autoscaling/custom_metrics_autoscaling.go

```diff
@@ -46,15 +47,32 @@ var _ = SIGDescribe("[HPA] Horizontal pod autoscaling (scale resource: Custom Me
 	})
 
 	f := framework.NewDefaultFramework("horizontal-pod-autoscaling")
-	var kubeClient clientset.Interface
 
-	It("should autoscale with Custom Metrics from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
-		kubeClient = f.ClientSet
-		testHPA(f, kubeClient)
+	It("should scale down with Custom Metric of type Pod from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
+		initialReplicas := 2
+		scaledReplicas := 1
+		deployment := monitoring.StackdriverExporterDeployment(stackdriverExporterDeployment, f.Namespace.Name, int32(initialReplicas), 100)
+		customMetricTest(f, f.ClientSet, podsHPA(f.Namespace.ObjectMeta.Name), deployment, nil, initialReplicas, scaledReplicas)
+	})
+
+	It("should scale down with Custom Metric of type Object from Stackdriver [Feature:CustomMetricsAutoscaling]", func() {
+		initialReplicas := 2
+		scaledReplicas := 1
+		deployment := monitoring.StackdriverExporterDeployment(dummyDeploymentName, f.Namespace.Name, int32(initialReplicas), 100)
```

Comment:
nit: 100 is pretty magical here, the intent is really 0.5 * 200 (which is hardcoded elsewhere). It would be pretty nice to replact 200 with a const and here use this const / 2.

Reply:
Fair enough, but if you don't mind I'll fix it with the next PR where I've done more refactoring and it will fit better there IMO.

Reply:
Sure, np. It's not particularly important anyway.